seqr/views/apis/igv_api.py

```diff
@@ -126,9 +128,44 @@ def fetch_igv_track(request, project_guid, igv_track_path):
     if igv_track_path.endswith('.bam.bai') and not does_file_exist(igv_track_path, user=request.user):
         igv_track_path = igv_track_path.replace('.bam.bai', '.bai')
 
+    if igv_track_path.startswith('gs://'):
+        return _stream_gs(request, igv_track_path)
+
     return _stream_file(request, igv_track_path)
 
 
+def _stream_gs(request, gs_path):
+    headers = {'Authorization': 'Bearer {}'.format(get_access_token())}
+    range_header = request.META.get('HTTP_RANGE')
+    if range_header:
+        headers['Range'] = range_header
+
+    bucket_object = gs_path[5:].split('/', 1)
+    for retry in range(2):  # retry automatically if the access token expired
```

Comment:
this is a messy way to ensure you have a valid access token, especially because its such a fast command to run and get the token. I would get rid of the global variable and just generate the access token as you need it

Reply:
Getting token with the 'gcloud auth' command is not fast because `subprocess` is slow. Each time when IGV loading a track, it needs tens of loading requests. We can't generate the access token for each of the requests. Another way to avoid retrying is to check the token expiration time.